entrypoint.go

```diff
@@ -0,0 +1,426 @@
+		}
+		defer cancel()
+		// start a goroutine to listen for cancellation file
+		go func() {
+			if err := e.waitingCancellation(ctx, cancel); err != nil && (!IsContextCanceledError(err) && !IsContextDeadlineError(err)) {
+				logger.Error("Error while waiting for cancellation", zap.Error(err))
+			}
+		}()
+		err = e.Runner.Run(ctx, e.Command...)
+	}
+
+	var ee *exec.ExitError
+	switch {
+	case err != nil && errors.Is(err, ErrContextCanceled):
+		logger.Info("Step was canceling")
+		output = append(output, e.outputRunResult(pod.TerminationReasonCancelled))
+		e.WritePostFile(e.PostFile, ErrContextCanceled)
+		e.WriteExitCodeFile(e.StepMetadataDir, syscall.SIGKILL.String())
+	case errors.Is(err, ErrContextDeadlineExceeded):
+		e.WritePostFile(e.PostFile, err)
+		output = append(output, e.outputRunResult(pod.TerminationReasonTimeoutExceeded))
+	case err != nil && e.BreakpointOnFailure:
+		logger.Info("Skipping writing to PostFile")
+	case e.OnError == ContinueOnError && errors.As(err, &ee):
+		// with continue on error and an ExitError, write non-zero exit code and a post file
+		exitCode := strconv.Itoa(ee.ExitCode())
+		output = append(output, result.RunResult{
+			Key:        "ExitCode",
+			Value:      exitCode,
+			ResultType: result.InternalTektonResultType,
+		})
+		e.WritePostFile(e.PostFile, nil)
+		e.WriteExitCodeFile(e.StepMetadataDir, exitCode)
+	case err == nil:
+		// if err is nil, write zero exit code and a post file
+		e.WritePostFile(e.PostFile, nil)
+		e.WriteExitCodeFile(e.StepMetadataDir, "0")
+	default:
+		// for a step without continue on error and any error, write a post file with .err
+		e.WritePostFile(e.PostFile, err)
+	}
+
+	// strings.Split(..) with an empty string returns an array that contains one element, an empty string.
+	// This creates an error when trying to open the result folder as a file.
+	if len(e.Results) >= 1 && e.Results[0] != "" {
+		resultPath := pipeline.DefaultResultPath
+		if e.ResultsDirectory != "" {
+			resultPath = e.ResultsDirectory
+		}
+		if err := e.readResultsFromDisk(ctx, resultPath, result.TaskRunResultType); err != nil {
+			logger.Fatalf("Error while handling results: %s", err)
+		}
+	}
+	if len(e.StepResults) >= 1 && e.StepResults[0] != "" {
+		stepResultPath := filepath.Join(e.StepMetadataDir, "results")
+		if e.ResultsDirectory != "" {
+			stepResultPath = e.ResultsDirectory
+		}
+		if err := e.readResultsFromDisk(ctx, stepResultPath, result.StepResultType); err != nil {
+			logger.Fatalf("Error while handling step results: %s", err)
+		}
+	}
+
+	if e.ResultExtractionMethod == config.ResultExtractionMethodTerminationMessage {
+		fp := filepath.Join(e.StepMetadataDir, "artifacts", "provenance.json")
+
+		artifacts, err := readArtifacts(fp)
+		if err != nil {
+			logger.Fatalf("Error while handling artifacts: %s", err)
+		}
+		output = append(output, artifacts...)
+	}
+
+	return err
+}
+
+func readArtifacts(fp string[]) ([]result.RunResult, error) {
```

Comment:
The function `readArtifacts` takes a string array as an argument but only uses the first element. If the function is not intended to handle multiple strings, consider changing the argument to a single string. If it should handle multiple strings, then the function implementation needs to be adjusted. [medium]